controllers/clusters/cadence_controller.go

```diff
@@ -73,44 +75,44 @@ func (r *CadenceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			logger.Info("Cadence resource is not found",
 				"resource name", req.NamespacedName,
 			)
-			return models.ExitReconcile, nil
+			return reconcile.Result{}, nil
 		}
 
 		logger.Error(err, "Unable to fetch Cadence resource",
 			"resource name", req.NamespacedName,
 		)
-		return models.ReconcileRequeue, nil
+		return reconcile.Result{}, err
 	}
 
 	switch cadenceCluster.Annotations[models.ResourceStateAnnotation] {
 	case models.CreatingEvent:
-		return r.HandleCreateCluster(ctx, cadenceCluster, logger), nil
+		return r.HandleCreateCluster(ctx, cadenceCluster, logger)
 	case models.UpdatingEvent:
-		return r.HandleUpdateCluster(ctx, cadenceCluster, logger), nil
+		return r.HandleUpdateCluster(ctx, cadenceCluster, logger)
 	case models.DeletingEvent:
-		return r.HandleDeleteCluster(ctx, cadenceCluster, logger), nil
+		return r.HandleDeleteCluster(ctx, cadenceCluster, logger)
 	case models.GenericEvent:
 		logger.Info("Generic event isn't handled",
 			"request", req,
 			"event", cadenceCluster.Annotations[models.ResourceStateAnnotation],
 		)
 
-		return models.ExitReconcile, nil
+		return reconcile.Result{}, nil
 	default:
 		logger.Info("Unknown event isn't handled",
 			"request", req,
 			"event", cadenceCluster.Annotations[models.ResourceStateAnnotation],
 		)
 
-		return models.ExitReconcile, nil
+		return reconcile.Result{}, nil
 	}
 }
 
 func (r *CadenceReconciler) HandleCreateCluster(
 	ctx context.Context,
 	cadence *v1beta1.Cadence,
 	logger logr.Logger,
-) reconcile.Result {
+) (ctrl.Result, error) {
```

Comment:
When we generate a controller with kubebuilder it creates default reconcile with ctrl.Result as returning value.
ctlr.Result is an alias for reconcile.Result. So, I decided to use the alias to avoid addition imports